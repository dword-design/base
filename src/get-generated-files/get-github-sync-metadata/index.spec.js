import { Base } from '@/src/index.js'

export default {
  'do not sync keywords'() {
    const base = new Base({ syncKeywords: false })
    expect(base.getGithubSyncMetadataConfig()).toMatchSnapshot(this)
  },
  valid() {
    const base = new Base()
    expect(base.getGithubSyncMetadataConfig()).toMatchSnapshot(this)
  },
  /**
   * Manual Tests
   *
   * - Creates branch action-sync-node-meta
   * - Changing the description updates the PR
   * - Changing the topics updates the PR
   */
}
