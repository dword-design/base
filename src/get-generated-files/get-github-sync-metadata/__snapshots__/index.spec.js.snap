// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`do not sync keywords 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v3",
        },
        Object {
          "uses": "jaid/action-sync-node-meta@v2.0.0",
          "with": Object {
            "approve": false,
            "commitMessage": "fix: write GitHub metadata to package.json [{changes}]",
            "githubToken": "\${{ secrets.GITHUB_TOKEN }}",
            "syncKeywords": false,
          },
        },
        Object {
          "uses": "gautamkrishnar/keepalive-workflow@v1",
        },
      ],
    },
  },
  "name": "sync-metadata",
  "on": Object {
    "schedule": Array [
      Object {
        "cron": "0 5 * * *",
      },
    ],
    "workflow_dispatch": Object {},
  },
}
`;

exports[`valid 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v3",
        },
        Object {
          "uses": "jaid/action-sync-node-meta@v2.0.0",
          "with": Object {
            "approve": false,
            "commitMessage": "fix: write GitHub metadata to package.json [{changes}]",
            "githubToken": "\${{ secrets.GITHUB_TOKEN }}",
          },
        },
        Object {
          "uses": "gautamkrishnar/keepalive-workflow@v1",
        },
      ],
    },
  },
  "name": "sync-metadata",
  "on": Object {
    "schedule": Array [
      Object {
        "cron": "0 5 * * *",
      },
    ],
    "workflow_dispatch": Object {},
  },
}
`;
